### ISSUE TITLE
clang-tidy Code Analysis error unless useBuildPath=true

### ISSUE THREAD

**Issue by richcarni** [microsoft/vscode-cpptools#9898]
I am using cpp and CMake tools, trying to keep everything as default as possible, but have run into an issue with my project when running Code Analysis. I get an error related to xmmintrin.h:

![image](https://user-images.githubusercontent.com/42241043/191143778-bdc0868d-7d3b-4fa6-8a3e-98cd217b3aba.png)

I've test that Code Analysis works OK with a basic CMake project

I then tested running clang-tidy directly from the command line with minimal compilation arguments (includes) and that also worked OK

Finally came to the workaround of setting useBuildPath=true:
```
"C_Cpp.codeAnalysis.clangTidy.useBuildPath": true,
"C_Cpp.default.compileCommands": "build/compile_commands.json",
```

So, not sure if this is a bug or expected behaviour? (Or me doing something stupid)

Not sure what arguments or how to check what arguments clang-tidy is being called with. I suspect that would shed some light on the issue

Happy to test further/provide more info as required

**Comment by sean-mcmanus** [comment]
Well, the errors in your screenshot are fixed relating to the __builtin_* types, but there might be more issues. Are you getting other errors after that? If so, do you know what the root C++ system header that your code is referencing that is pulling in the xmmintrin.h? And is your repro on Windows with mingw or Linux?

**Comment by sean-mcmanus** [comment]
Actually, my testing was incorrect and it's not fixed. I need to know how the info previously mentioned in order tell if this is fixable or not.

**Comment by richcarni** [comment]
The only error I get is related to xmmintrin.h, no errors after that

I don't know which library is pulling it in. I'm working on a numerical problem using Ceres Solver and Adept. I know Ceres has a bunch of dependencies including Eigen, ATLAS and others. Sounds like you need to where it is being included to investigate further - it doesn't seem straightforward, but I'll look into it

I am running on Linux, gcc 11.2.0

**Comment by richcarni** [comment]
It's [Eigen](https://eigen.tuxfamily.org/index.php?title=Main_Page)

Eigen/Core -> src/Core/util/ConfigureVectorization.h -> xmmintrin.h

Let me know what further info I can provide

**Comment by sean-mcmanus** [comment]
Okay, thanks, that's what I was looking for.

**Comment by brosenberg42** [comment]
I am having the same problem except that when I add:
```
"C_Cpp.codeAnalysis.clangTidy.useBuildPath": true,
"C_Cpp.default.compileCommands": "build/compile_commands.json",
```
I get "'stddef.h' file not found". I tried setting `export CPATH=/usr/lib/gcc/x86_64-linux-gnu/9/include`, but that resulted in getting the __builtin errors again.
I am using gcc 9.4.0 on Ubuntu 20.04.

**Comment by sean-mcmanus** [comment]
@brosenberg42 You need to set C_Cpp.codeAnalysis.clangTidy.path to the clang-tidy installed on your machine.

**Comment by sean-mcmanus** [comment]
@richcarni I've verified that the fix had made a few weeks ago fixes the issue with the Eigen library and gcc. It should be in our next release.

**Comment by richcarni** [comment]
Thanks @sean-mcmanus - will be sure to test in next release and report back

**Comment by richcarni** [comment]
Unfortunately not having any better luck with v1.13.2. Still getting error citing xmmintrin.h

**Comment by sean-mcmanus** [comment]
Ah, sorry about that -- looks like my verification process had a flaw.

I figured out a fix. You can workaround it via using
```
    "C_Cpp.codeAnalysis.clangTidy.args": [
        "--extra-arg=-mno-sse"
    ]
```

**Comment by sean-mcmanus** [comment]
Fixed with 1.13.4 (pre-release): https://github.com/microsoft/vscode-cpptools/releases/tag/v1.13.4

**Comment by richcarni** [comment]
Can confirm this is now working for me! (1.13.4)

Thanks @sean-mcmanus for the updates
